libraries/AP_HAL_PX4/CAN.cpp

```diff
@@ -256,31 +256,43 @@ int PX4CAN::computeTimings(const uint32_t target_bitrate, Timings& out_timings)
     // First attempt with rounding to nearest
     BsPair solution(bs1_bs2_sum, uint8_t(((7 * bs1_bs2_sum - 1) + 4) / 8));
 
-    if (solution.sample_point_permill > MaxSamplePointLocation) {
-        // Second attempt with rounding to zero
-        solution = BsPair(bs1_bs2_sum, uint8_t((7 * bs1_bs2_sum - 1) / 8));
-    }
+    if (solution.isValid())
+    {
+        if (solution.sample_point_permill > MaxSamplePointLocation) {
```

Comment:
I'm not sure how this is better than before. Maybe I didn't explain properly but what I proposed was to use the second solution if the first wasn't valid: it was simply adding to the previous if `|| !solution.isValid()`, so it would be:

 `if (solution.sample_point_permill > MaxSamplePointLocation || !solution.isValid())`

These changes actually seem worse, now it won't use the second solution when the first is invalid even if `sample_point_permill > MaxSamplePointLocation`.